lms/services/lti_role_service.py

```diff
@@ -23,6 +23,13 @@ def get_roles(self, role_description: str) -> List[LTIRole]:
         # Pylint is confused about the `in_` for some reason
         roles = self._db.query(LTIRole).filter(LTIRole.value.in_(role_strings)).all()
 
+        for role in roles:
+            # LTIRole.value is a setter so this has the side effect changing
+            # both role.type and role.scope.
+            # This is useful when the logic for those fields change, updating the
+            # values in the DB and also exposing the right values in the rest to the application.
```

Comment:
```suggestion
            # LTIRole.value is a setter so this has the side effect changing
            # both role.type and role.scope.
            # This is useful when the logic for those fields change, updating 
            # the values in the DB and also exposing the right values in the 
            # rest to the application.
```

If we didn't want to use a side effect like this we could change the model object to be like this:

```python
    @value.setter
    def value(self, value):
        self._value = value

        self.update_from_value()

    def update_from_value(self):
        # pylint: disable=protected-access
        self.scope, self.type = _RoleParser._parse_role(self._value)
```

Then we could call the `update_from_value()` public method, rather than rely on it as a side effect.

Also looking at this now, having a private method on a private object doesn't seem right? Should `_parse_role` be public?